validator_manager/src/delete_validator.rs

```diff
@@ -0,0 +1,266 @@
+use std::path::PathBuf;
+
+use clap::{Arg, ArgAction, ArgMatches, Command};
+use eth2::{
+    lighthouse_vc::types::{DeleteKeystoreStatus, DeleteKeystoresRequest},
+    SensitiveUrl,
+};
+use serde::{Deserialize, Serialize};
+use types::PublicKeyBytes;
+
+use crate::{common::vc_http_client, DumpConfig};
+
+pub const CMD: &str = "delete";
+pub const VC_URL_FLAG: &str = "vc-url";
+pub const VC_TOKEN_FLAG: &str = "vc-token";
+pub const VALIDATOR_FLAG: &str = "validator";
+
+#[derive(Debug)]
+pub enum DeleteError {
+    InvalidPublicKey,
+    DeleteFailed(eth2::Error),
+}
+
+pub fn cli_app() -> Command {
+    Command::new(CMD)
+        .about("Deletes validator from a validator client using the HTTP API.")
+        .arg(
+            Arg::new(VC_URL_FLAG)
+                .long(VC_URL_FLAG)
+                .value_name("HTTP_ADDRESS")
+                .help(
+                    "A HTTP(S) address of a validator client using the keymanager-API. \
+            If this value is not supplied then a 'dry run' will be conducted where \
+            no changes are made to the validator client.",
+                )
+                .default_value("http://localhost:5062")
+                .requires(VC_TOKEN_FLAG)
+                .action(ArgAction::Set)
+                .display_order(0),
+        )
+        .arg(
+            Arg::new(VC_TOKEN_FLAG)
+                .long(VC_TOKEN_FLAG)
+                .value_name("PATH")
+                .help("The file containing a token required by the validator client.")
+                .action(ArgAction::Set)
+                .display_order(0),
+        )
+        .arg(
+            Arg::new(VALIDATOR_FLAG)
+                .long(VALIDATOR_FLAG)
+                .value_name("STRING")
+                .help("Validator that will be deleted (pubkey).")
+                .action(ArgAction::Set)
+                .display_order(0),
+        )
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub struct DeleteConfig {
+    pub vc_url: SensitiveUrl,
+    pub vc_token_path: PathBuf,
+    pub validator_to_delete: PublicKeyBytes,
```

Comment:
I wonder if it is worth supporting being able to delete vals in bulk. Maybe just starting with supporting comma separation in the CLI for now. Later maybe we could support passing in a file with them in a list or something.

Reply:
Added the support for comma separated to delete more than 1 validator in [4726501](https://github.com/sigp/lighthouse/pull/6261/commits/472650108dc599ee47a3aa612d5430059a980f50) with a lot of help from @michaelsproul, thank you!